### PR TITLE
Add CSI v2.4.1

### PR DESCRIPTION
**Pull request by doflamingo721**
Adds:
CSI v2.4.1
Porting `permit-os` from https://github.com/rancher/charts/pull/1755

Linked issue:
https://github.com/rancher/rancher/issues/33848
https://github.com/rancher/rancher/issues/36081

### CODE REVIEW

File: charts/rancher-vsphere-csi/Chart.yaml [hunk 1]
```diff
@@ -6,16 +6,16 @@ annotations:
   catalog.cattle.io/rancher-version: '>= 2.6.0-0 <= 2.6.99-0'
```
Comment (PennyScissors) [inline charts/rancher-vsphere-csi/Chart.yaml:6]: Update this to `catalog.cattle.io/rancher-version: '>= 2.6.0-0 <= 2.6.100-0'`, and add catalog.cattle.io/kube-version: '>= 1.19.0-0 <= 1.23.0-0'.

File: charts/rancher-vsphere-csi/Chart.yaml [hunk 2]
```diff
@@ -6,16 +6,16 @@ annotations:
...
   catalog.cattle.io/release-name: vsphere-csi
 apiVersion: v1
-appVersion: 2.3.0
+appVersion: 2.4.1
 description: vSphere Cloud Storage Interface (CSI)
 icon: https://charts.rancher.io/assets/logos/vsphere-csi.svg
 keywords:
 - infrastructure
-kubeVersion: 1.19 - 1.21
+kubeVersion: 1.19 - 1.22
```
Comment (PennyScissors) [inline charts/rancher-vsphere-csi/Chart.yaml:14]: Delete this line entirely, the kube-version annotation replaces this.

File: charts/rancher-vsphere-csi/templates/node/role-binding.yaml [hunk 3]
```diff
@@ -12,3 +12,18 @@ roleRef:
...
+---
+
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: vsphere-csi-node-cluster-role-binding
+subjects:
+  - kind: ServiceAccount
+    name: vsphere-csi-node
+    namespace: vmware-system-csi
```
Comment (PennyScissors) [inline charts/rancher-vsphere-csi/templates/node/role-binding.yaml:25]: should be `namespace: {{ .Release.Namespace }}`

File: charts/rancher-vsphere-csi/Chart.yaml [hunk 4]
```diff
@@ -6,16 +6,16 @@ annotations:
   catalog.cattle.io/rancher-version: '>= 2.6.0-0 <= 2.6.99-0'
   catalog.cattle.io/release-name: vsphere-csi
 apiVersion: v1
-appVersion: 2.3.0
+appVersion: 2.4.1
```
Comment (PennyScissors) [inline charts/rancher-vsphere-csi/Chart.yaml:9]: Can we make this `2.4.1-rancher1` to match the `version`?

File: charts/rancher-vsphere-csi/templates/node/daemonset.yaml [hunk 5]
```diff
@@ -18,137 +18,128 @@ spec:
         app: vsphere-csi-node
         role: vsphere-csi
     spec:
+      nodeSelector:
+        kubernetes.io/os: linux
```
Comment (PennyScissors) [inline charts/rancher-vsphere-csi/templates/node/daemonset.yaml:22]: Template this like [this](https://github.com/rancher/charts/blob/31f56d0b1c01592b70dd9763f1aadfed53e78572/charts/rancher-vsphere-csi/100.1.0%2Bup2.3.0/templates/controller/deployment.yaml#L20-L24), but for `csiNode` in values.yaml. Also, I do we need affinity/tolerations for rke and rke2 like we do for the deployment [here](https://github.com/rancher/charts/blob/31f56d0b1c01592b70dd9763f1aadfed53e78572/charts/rancher-vsphere-csi/100.1.0%2Bup2.3.0/templates/controller/deployment.yaml#L25-L62)? You may want to test that.

File: charts/rancher-vsphere-csi/Chart.yaml (hunk 4 above)
Comment (doflamingo721) [reply charts/rancher-vsphere-csi/Chart.yaml]: Updated the `appVersion` to match `version`

File: charts/rancher-vsphere-csi/templates/node/role-binding.yaml (hunk 3 above)
Comment (doflamingo721) [reply charts/rancher-vsphere-csi/templates/node/role-binding.yaml]: Made the required changes.

File: charts/rancher-vsphere-csi/templates/node/daemonset.yaml (hunk 5 above)
Comment (doflamingo721) [reply charts/rancher-vsphere-csi/templates/node/daemonset.yaml:27]: Added the nodeSelector template

**Review by PennyScissors — APPROVED** [review]

**Pull request merged**